### DISCUSSION TITLE
ImageMagick not working with Moviepy

### DISCUSSION THREAD

**Discussion by Cavicool** [ImageMagick/ImageMagick#7022] [Help] [answered]
### ImageMagick version

7.1.1-26-Q16

### Operating system, version and so on

Windows 11 

### Description

I'm currently using MoviePy for a project, and it told me to install ImageMagick. I've done that and changed the `conf.py` file in MoviePy using this code:

```python
from moviepy.config import change_settings
change_settings({"IMAGEMAGICK_BINARY": r"C:\ProgramData\Microsoft\Windows\Start Menu\Programs\ImageMagick 7.1.1 Q16-HDRI (64-bit)"})
```

However, it gave me an error and still told me:

_**This error can be due to the fact that ImageMagick is not installed on your computer, or (for Windows users) that you didn't specify the path to the ImageMagick binary in file `conf.py`, or that the path you specified is incorrect.**_

I asked Bing Chat, and it suggested that I need to use a different path with a file named `magick.exe` located in a directory named "bin." I searched for the folder in the ImageMagick file but couldn't find it, even after trying to reinstall it multiple times. How can I access this folder or is there another way to resolve this issue?

**Comment by fmw42** [comment]
What was the exact error?  Have you installed the correct version of Imagemagick for your OS and also which version (Imagemagick 6 or Imagemagick7) does Moviepy require. It might not work with Imagemagick 7 and want Imagemagick 6.

**Reply by Cavicool** [reply]
entire error:

Traceback (most recent call last):
  File "C:\Users\username\AppData\Local\Programs\Python\Python311\Lib\site-packages\moviepy\video\VideoClip.py", line 1137, in __init__
    subprocess_call(cmd, logger=None)
  File "C:\Users\username\AppData\Local\Programs\Python\Python311\Lib\site-packages\moviepy\tools.py", line 46, in subprocess_call
    proc = sp.Popen(cmd, **popen_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\username\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\username\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 1538, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
PermissionError: [WinError 5] Zugriff verweigert

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\username\Downloads\Video_Generator\script.py", line 122, in <module>
    subtitle = TextClip(text, font="Arial", color="white", fontsize=24, method="caption", align="center", size=clip.size)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\username\AppData\Local\Programs\Python\Python311\Lib\site-packages\moviepy\video\VideoClip.py", line 1146, in __init__
    raise IOError(error)
OSError: MoviePy Error: creation of None failed because of the following error:

[WinError 5] acces denied.

**Reply by fmw42** [reply]
this error message does not seem to involve Imagemagick. It is a Python error.  Why do you think it has to do with Imagemagick?  It looks like a Windows error about permissions or where you are trying to do the Moviepy install.

**Reply by Cavicool** [reply]
i thought that because it was still printing out the message with the conf.py thing but it i will try running the script as adminstrator if that isnt the default(running as administrator changed nothing)

**Reply by Cavicool** [reply]
I also found someone having a similar issue and he resolved it by putting that path into config.py file but i dont have the magick.exe file so i cant do it

https://stackoverflow.com/questions/65136644/getting-error-about-imagemagick-with-python-moviepy-when-i-try-add-text-clip#:~:text=MoviePy%20may%20need%20IM%206%20or%20it%20may,legacy%20components%20from%20the%20install%20window.%20%E2%80%93%20fmw42

**Comment by snibgo** [comment]
> C:\ProgramData\Microsoft\Windows\Start Menu\Programs\ImageMagick 7.1.1 Q16-HDRI (64-bit)

Does that directory contain magick.exe?

**Reply by Cavicool** [reply]
no it doesnt i added that because i thought that was the magic.exe but it doesnt work magic.exe is something different
but for some reason the installation doesnt contain that file

**Comment by snibgo** [comment]
>  but i dont have the magick.exe file so i cant do it

You have a Windows system. The main ImageMagick v7 program is `magick.exe`. So, you haven't installed ImageMagick? If you think you did, then what did you install it from? And where did you install it to (what directory on your computer)?

As fmw42 says, perhaps Moviepy need ImageMagick v6. For that, the main program is `convert.exe`.

**Reply by Cavicool** [reply]
I installed it from the website and the directory is programms

**Reply by Cavicool** [reply]
but do you know how i acces the magick.exe file ?

**Comment by snibgo** [comment]
> I installed it from the website...

What website?

**Reply by Cavicool** [reply]
the Image Magick website sorry that i didn't clarify myself

**Reply by Cavicool** [reply]
Do you know if theres is a bin folder when you download Image Magick or a magick.exe file ?

**Comment by fmw42** [comment]
What happens if you run

`magick.exe -version`

**Reply by Cavicool** [reply]
My results:

Version: ImageMagick 7.1.1-26 Q16-HDRI x64 83eefaf:20240107 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC HDRI Modules OpenCL OpenMP(2.0) 
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib

**Reply by fmw42** [reply]
So Imagemagick 7 is installed.  But does Moviepy know where it was installed?  Does Moviepy need Imagemagick 6?

**Reply by Cavicool** [reply]
i dont think it needs  Imagemagick 6. But Moviepy definetly doesnt know if ImageMagick is installed or where it is. I tried to specifiy that in the path in the line of code i wrote but it cant find it because moviepy needs to have the magic.exe file in the part I think and that doesnt exist

**Comment by snibgo** [comment]
What is the text output from this command:
```
where magick.exe
```

**Reply by Cavicool** [reply]
no result nothing happens

**Comment by Cavicool** [comment, marked as answer]
I used a function to get more information and found out exactly what path i need to paste into that line of code. I missunderstood magick.exe(convert.exe) for some sort of file but its just some text that is used to represent the programm. Thanks for trying to help and respoding so fast.

**Reply by bharathreddyza** [reply]
what was the solution?

**Reply by bfrasure** [reply]
look in the config_defaults.py file. It specifies what it's looking for.  under IMAGEMAGICK_BINARY
